tool/gravity/cli/operation.go

```diff
@@ -116,24 +117,44 @@ func completeOperationPlan(localEnv, updateEnv, joinEnv *localenv.LocalEnvironme
 	switch op.Type {
 	case ops.OperationInstall:
 		// There's only one install operation
-		return completeInstallPlan(localEnv, op)
+		err = completeInstallPlan(localEnv, op)
 	case ops.OperationExpand:
-		return completeJoinPlan(localEnv, joinEnv, op)
+		err = completeJoinPlan(localEnv, environ, op)
 	case ops.OperationUpdate:
-		return completeUpdatePlan(localEnv, updateEnv, op)
+		err = completeUpdatePlan(localEnv, environ, op)
 	case ops.OperationUpdateRuntimeEnviron:
-		return completeEnvironPlan(localEnv, updateEnv, op)
+		err = completeEnvironPlan(localEnv, environ, op)
 	case ops.OperationUpdateConfig:
-		return completeConfigPlan(localEnv, updateEnv, op)
+		err = completeConfigPlan(localEnv, environ, op)
 	default:
 		return trace.BadParameter("operation type %q does not support plan completion", op.Type)
 	}
+	if op.Type != ops.OperationInstall && trace.IsNotFound(err) {
```

Comment:
Is this special use-case mostly to be able to complete join (or any operation?) from another node?

Reply:
Yes.